modules/bim/spec/features/ifc_models/ifc_upload_shared_examples.rb

```diff
@@ -2,25 +2,42 @@
   let(:user) { FactoryBot.create :admin }
   let(:project) { FactoryBot.create :project, enabled_module_names: %i[bim] }
   let(:ifc_fixture) { ::UploadedFile.load_from('modules/bim/spec/fixtures/files/minimal.ifc') }
+  let(:set_tick_is_default_after_file) { true }
 
   before do
     login_as user
 
     allow_any_instance_of(Bim::IfcModels::BaseContract).to receive(:ifc_attachment_is_ifc).and_return true
   end
+  
+  shared_examples 'should allow uploading an IFC file' do
+    it 'should allow uploading an IFC file' do
+      visit new_bcf_project_ifc_model_path(project_id: project.identifier)
 
-  it 'should allow uploading an IFC file' do
-    visit new_bcf_project_ifc_model_path(project_id: project.identifier)
+      page.find('#bim_ifc_models_ifc_model_is_default').set(true) if set_tick_is_default_after_file
+      page.attach_file("file", ifc_fixture.path, visible: :all)
+      page.find('#bim_ifc_models_ifc_model_is_default').set true unless set_tick_is_default_after_file
```

Comment:
I feel like this flag should be used the other way round. I.e. `unless` in line 17 and `if` in line 19. For the rest result it doesn't matter since both cases are tested anyway and both expect the same result. Still, this is confusing so I suggest you swap this and everything else looks fine!